pkg/middleware/database_logging.go

```diff
@@ -43,10 +43,11 @@ func (dlm DatabaseLoggingMiddleware) Handler(next http.Handler) http.Handler {
 			return
 		}
 
-		d.LogMode(true)
-		d.SetLogger(newGormLogger(l))
+		newDb := d.New()
```

Comment:
If we have to create a new instance (clone) then I think the `database` middleware is a better place (instead of the database logging middleware). For example in `pkg/middleware/database.go`:

 ```go
// Handler implements the middlewares.Handlerer interface: it returns a
// http.Handler to be mounted as middleware. The handler injects the database
// connection pool to the request context.
func (dm DatabaseMiddleware) Handler(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		db := sdkinstrumentation.TraceDatabase(r.Context(), dm.Database.New()) // Added .New()
		ctx := sdkdatabasecontext.ToContext(r.Context(), db)
		next.ServeHTTP(w, r.WithContext(ctx))
	})
}
```

Reply:
@fotos The whole point of this change is to make this middleware self-contained and be dependant on another side-effect. We implicitly create a new DB in the `database` middleware already, but we shouldn't rely and depend on another middleware

Reply:
Got it.

My line of thinking was to move the responsibility for the middleware "uniqueness" to the DB setup: in other words the `DatabaseLogging{Middleware,Interceptor}` would assume that the DB it gets from the context is already "unique" (scoped to the request) and use it.

I can see this going both ways, so I'll resolve the thread and leave it up to you. 😊